### PR TITLE
Upgrade to fc34

### PR DESCRIPTION
**Pull request by rleon**

### CODE REVIEW

**Comment by rleon** [comment]
It is a false failure due to the missing FC34 image in azure.

**Comment by rleon** [comment]
@jgunthorpe can you please push such image?

Thanks

File: libibmad/register.c [hunk 1]
```diff
@@ -117,15 +117,15 @@ int mad_register_client_via(int mgmt, uint8_t rmpp_version,
 }
 
 int mad_register_server(int mgmt, uint8_t rmpp_version,
-			long method_mask[], uint32_t class_oui)
+			long method_mask[16 / sizeof(long)], uint32_t class_oui)
```
Comment (jgunthorpe) [inline libibmad/register.c:120]: Why not just use "long method_mask *" ?

File: buildlib/azure-pipelines.yml [hunk 2]
```diff
@@ -34,7 +34,7 @@ resources:
       image: ucfconsort.azurecr.io/rdma-core/centos8:25.0
       endpoint: ucfconsort_registry
     - container: fedora
-      image: ucfconsort.azurecr.io/rdma-core/fc33:33.0
+      image: ucfconsort.azurecr.io/rdma-core/fc34:34.0
```
Comment (jgunthorpe) [inline buildlib/azure-pipelines.yml:37]: Change the descriptive string below as well

**Comment by jgunthorpe** [comment]
Hum those new warnings are interesting, they are bogus becuase 'hdr' is enclosed in a larger object which is the thing we are actually trying to write. I wonder if a void * cast would silence it?

File: libibmad/register.c (hunk 1 above)
Comment (rleon) [reply libibmad/register.c:120]: It still produces same warning of differences between .c and .h. We can't change .h because it is ABI.

**Comment by rleon** [comment]
Interesting why can't I reproduce CI failures locally?

**Comment by jgunthorpe** [comment]
We don't run the fc packaging build with -Werror outside AZP. I think the rpm scripts are injecting some addtional compile options compared to our normal make. Usually what I try to do is pull the extra options into the cmake so everything turns them on

**Comment by rleon** [comment]
> We don't run the fc packaging build with -Werror outside AZP. I think the rpm scripts are injecting some addtional compile options compared to our normal make. Usually what I try to do is pull the extra options into the cmake so everything turns them on

I tried, but maybe not enough.

**Comment by rleon** [comment]
> > We don't run the fc packaging build with -Werror outside AZP. I think the rpm scripts are injecting some addtional compile options compared to our normal make. Usually what I try to do is pull the extra options into the cmake so everything turns them on
> 
> I tried, but maybe not enough.

ok, the flag that reproduces it is "-flto=auto"

**Pull request merged**